ocs_ingester/ingester.py

```diff
@@ -56,12 +59,13 @@ def frame_exists(fileobj, api_root=settings.API_ROOT, auth_token=settings.AUTH_T
 
     """
     archive = ArchiveService(api_root=api_root, auth_token=auth_token)
-    md5 = File(fileobj, run_validate=False).get_md5()
+    md5 = get_md5_and_collect_metrics(File(fileobj))
     return archive.version_exists(md5)
 
 
-def validate_fits_and_create_archive_record(fileobj, path=None, file_metadata=None, required_headers=settings.REQUIRED_HEADERS,
-                                            blacklist_headers=settings.HEADER_BLACKLIST):
+def validate_fits_and_create_archive_record(fileobj, path=None, file_metadata=None,
+                                            required_headers=archive_settings.REQUIRED_HEADERS,
+                                            blacklist_headers=archive_settings.HEADER_BLACKLIST):
```

Comment:
While we're in here changing a whole lot of stuff, might as well consider changing `HEADER_BLACKLIST` to `HEADER_DENYLIST`

Reply:
I agree with Elisabeth here - it'd be easier now to do this than to do it later, and this is an open source project.

https://abusix.com/resources/blocklists/why-changing-the-terms-blacklist-and-whitelist-isnt-as-easy-as-it-might-seem/

Reply:
Doing a search in the LCOGT org - it appears that we don't override this value anywhere

https://github.com/search?q=org%3ALCOGT+HEADER_BLACKLIST

Reply:
oh okay, I don't think I understood what she was saying before but I get it. Yeah, I think we could update this to denylist fine...